Sage/carbon
# fix(scrollable-block): ensure border-styling is applied when appropriate to do so

### Proposed behaviour



Border radius should be 0px on all corners when the last menu item does not exceed the bottom of the scrollable-block or the submenu itself. 0px, 0px, 0px, 8px should only be applied when the last menu item exceeds the bottom of the submenu block. 

![Screenshot 2024-04-08 082650](https://github.com/Sage/carbon/assets/101639841/0b3d08ee-6df7-4dff-8a18-fe706ea6faba)
![Screenshot 2024-04-08 082719](https://github.com/Sage/carbon/assets/101639841/b5269ac1-5904-4aef-8032-8dbe79450102)


### Current behaviour

border-radius styling was being set incorrectly

![previous behaviour Screenshot 2024-04-08 083135](https://github.com/Sage/carbon/assets/101639841/3b5ecf12-513e-4c77-a78b-70ab72a6f237)




### Checklist




#### QA


### Additional context



### Testing instructions


- Go to the [Scrollable submenu with parent](https://carbon.sage.com/?path=/docs/menu--scrollable-submenu-with-parent)
- Open the Submenu (Menu Item Three)
- In the Search component type ap and you see it returns two results. apple and grapefruit
- grapefruit has incorrect styling as you’ll be able to see the bottom left border has a 8px radius. It should be 0px in this case.
- Only if the scrollbar is present should the last menu item in the scrollable block have an 8px bottom left border radius applied.
